app/jobs/gpo_expiration_job.rb

```diff
@@ -1,9 +1,10 @@
 class GpoExpirationJob < ApplicationJob
   queue_as :low
 
-  def initialize(analytics: nil, on_profile_expired: nil)
+  def initialize(*args, analytics: nil, on_profile_expired: nil, **rest)
     @analytics = analytics
     @on_profile_expired = on_profile_expired
+    super(*args, **rest)
```

Comment:
What's the actual method signature for the base constructor? Is is this?

https://github.com/rails/rails/blob/62b1cec/activejob/lib/active_job/core.rb#L95

The reason I ask is (1) to make sure we're passing only what it's expecting, and (2) to make sure the absence of a few of the `rest` properties that we're explicitly picking (`analytics` and `on_profile_expired`) oughtn't be passed as well.

If it is just the one positional argument splat, maybe it suffices to pass just `args`?

```suggestion
  def initialize(*args, analytics: nil, on_profile_expired: nil)
    @analytics = analytics
    @on_profile_expired = on_profile_expired
    super(*args)
```

Reply:
I think in general we're only supposed to define a `perform` method

Reply:
Yeah, you might be right. TBH, I copied the pattern from [the `MonthlyKeyMetricsReport` job](https://github.com/18F/identity-idp/blob/main/app/jobs/reports/monthly_key_metrics_report.rb#L10). I'm going to run a quick test and see if I can get away with not including `**rest` here.

Reply:
Yeah I can probably rework this to get rid of `initialize` entirely